### DISCUSSION TITLE
Issues with version 4

### DISCUSSION THREAD

**Discussion by van-vince** [refinedev/refine#3902] [Q&A] [answered]
I upgraded my project to refine version 4  and set up the routing by the resources are not showing on the sidebar even though the routes are working when i change in the browser. 
![Screenshot 2023-03-17 at 2 11 26 AM](https://user-images.githubusercontent.com/67305320/225796600-09159813-f935-41a5-b3fa-28b4f405dad4.png)

**Comment by aliemir** [comment, marked as answer]
Hey @van-vince sorry for the issue! 

Are you using a custom `<Sider>`component? Looks like an issue related to the router changes.

If you're using the new `routerProvider` prop; you'll also need to update the usage of `useRouterContext` hook, which is deprecated and only works with the legacy router providers.

You can replace the `Link` usage from `useRouterContext` with the `useLink` hook;

```diff
- import { useRouterContext } from "@refinedev/core";
+ import { useLink } from "@refinedev/core";

const CustomSider = () => {
-   const { Link } = useRouterContext();
+   const Link = useLink();

    /* ... */
}
```

Let me know if this resolved the issue, if not can you give us a bit more info about the issue? Let's try to understand the issue. 🙏

**Reply by van-vince** [reply]
Thank you very much, all the resources are showing now
